packages/react/src/analytics/integrations/GA4/GA4.ts

```diff
@@ -48,96 +45,99 @@ import getDefaultCommandsBuilder, {
   nonInteractionEvents,
 } from './commands';
 import merge from 'lodash/merge';
+import type {
+  EventScopeCommandsHandlers,
+  GA4CommandList,
+  GA4IntegrationOptions,
+  NonInteractionEvents,
+  OnPreProcessCommandsHandler,
+  ProductMappings,
+  Schemas,
+  ScopeCommands,
+} from './types';
 
 /**
  * Google Analytics 4 Integration.
- *
- * @private
- * @augments Integration
  */
-class GA4 extends integrations.Integration {
-  enableAutomaticPageViews;
-  schemaEventsMap;
-  initializePromise;
-  initializePromiseResolve;
-  optionsValidationResultsMap;
+class GA4 extends integrations.Integration<GA4IntegrationOptions> {
+  private enableAutomaticPageViews!: boolean;
+  private schemaEventsMap!: Schemas;
+  private initializePromise!: Promise<void>;
+  private initializePromiseResolve?:
+    | ((value?: void | Promise<void>) => void)
+    | null;
+  private optionsValidationResultsMap!: Map<string, boolean>;
+  private measurementId!: string;
+  private scopeCommands!: ScopeCommands;
+  private productMappings!: ProductMappings;
+  private nonInteractionEvents!: NonInteractionEvents;
+  private onPreProcessCommands?: OnPreProcessCommandsHandler;
+  private setCustomUserIdProperty!: unknown;
+  private customLoadScriptFn?: () => Promise<void> | undefined;
 
   /**
    * Creates an instance of Google Analytics 4 integration.
    * Setup Google Analytics 4 and initializes event commands map that will be used
    * to obtain the command list associated with an event.
    *
-   * @param {object} options - User configured options.
-   * @param {object} loadData - Analytics's load event data.
-   * @param {object} analytics - Analytics stripped down instance.
+   * @param options - User configured options.
+   * @param loadData - Analytics's load event data.
+   * @param strippedDownAnalytics - Analytics stripped down instance.
    *
    */
-  constructor(options, loadData, analytics) {
-    super(options, loadData, analytics);
+  constructor(
+    protected options: GA4IntegrationOptions,
+    protected loadData: LoadIntegrationEventData,
+    protected strippedDownAnalytics: StrippedDownAnalytics,
+  ) {
+    super(options, loadData, strippedDownAnalytics);
     this.initialize(options);
     this.onSetUser(loadData);
   }
 
   /**
    * Method to check if the integration is ready to be loaded.
    *
-   * @param {object} consent - The consent object representing the user preferences.
+   * @param consent - The consent object representing the user preferences.
    *
-   * @returns {boolean} If the integration is ready to be loaded.
+   * @returns If the integration is ready to be loaded.
    */
-  static shouldLoad(consent) {
+  static shouldLoad(consent: ConsentData | null): boolean {
```

Comment:
```suggestion
  static shouldLoad(consent: ConsentData): boolean {
```
(I know that we check anyway if it is defined or not to be extra careful, but the `shouldLoad` function should only make sense if consent parameter is set to a valid value).